doc/release/upcoming_changes/23136.performance.rst

```diff
@@ -0,0 +1,17 @@
+``ufunc.at`` can be much faster
+-------------------------------
+Generic ``ufunc.at`` can be up to 9x faster. The conditions for this speedup:
+
+- contiguous arguments
+- no casting
+
+If ufuncs with appropriate indexed loops on 1d arguments with the above
+conditions, ``ufunc.at`` can be up to 60x faster (an additional 7x speedup).
+Appropriate indexed loops have been added to ``add``, ``subtract``,
+``multiply``, ``divide`` (and ``floor_divide``)
+
+The internal logic is similar to the logic used for regular ufuncs, which also
+have a fast path for contiguous, non-casting, 1d arrays.
```

Comment:
1-D really doesn't matter much for ufuncs (although broadcasting is limited to 0-D arrays).

In any case, I am bike-shedding.  Its fine as is to get the gist.